### ISSUE TITLE
[Deque Analysis] ARIA / Label Issues in Submission form

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1272]
> Critical: This issue results in blocked content for people with disabilities, and will definitely prevent them from accessing fundamental features or content. This type of issue puts your organization at risk. Prioritize fixing as soon as possible, within the week if possible. Remediation should be a top priority. 

## Deque Analysis Summary

Two remaining Submission form critical issues exist.  Both of these seem to be caused by external libraries.

1. (Issue ID: 469503) The `aria-multiline` attribute is invalid on `combobox` role (it currently appears for `dc.subject` field). See https://www.w3.org/TR/wai-aria-1.1/#aria-multiline  It should be removed.
    * the aria is added by the [ngbTypeahead component,](https://ng-bootstrap.github.io/#/components/typeahead/examples)
2. (Issue ID: 469538) The `<select>` for the `dc.identifier` field doesn't have a `<label>` or an `aria-label` attribute. 
    * That element is added by the form library

The full list of these issues (which includes links for more info) can be found at this link (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?activeTab=dt-issue&page=0&pageSize=100&sortField=ordinal&sortDir=asc&row=9&filter%5Bseverity%5D=4&filter%5Bpage_number%5D=10&filter%5Bpage_number%5D=11

## More Information / Tools

* Consider using [Deque's free Chrome plugin](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) to check your work

**Comment by atarix83** [comment]
@tdonohue 

I've opened an issue to ng-bootstrap, since I don't see other ways to fix this issue  

https://github.com/ng-bootstrap/ng-bootstrap/issues/4181

**Comment by tdonohue** [comment]
@atarix83 : Thanks for letting me know.  In that case, we'll wait to see if there's response from ng-bootstrap on how to fix that first issue.  If we need to, we can separate that out into a different ticket & make it a lower priority, since it's not something we can take action on.

In that case, I'd just recommend seeing if we can solve 
`(Issue ID: 469538) The <select> for the dc.identifier field doesn't have a <label> or an aria-label attribute.`

**Comment by tdonohue** [comment]
@atarix83 : thanks, that sounds good to me.  I'll assign to you for 4Science to work on.

**Comment by tdonohue** [comment]
In today's Developers Meeting, @atarix83 noted that **resolving this ticket requires upgrading `ng-bootstrap` to version 13.2.0** (as that is the version where https://github.com/ng-bootstrap/ng-bootstrap/issues/4181 was fixed).

However, we cannot upgrade to `ng-bootstrap` v13 because `@ng-dynamic-forms` is incompatible with `ng-bootstrap` v13.  In other words, #2216 is "blocking" us from upgrading to `ng-bootstrap` v13.

As a workaround, @atarix83 proposed we *temporarily* do the following (described in [phase 1 of this comment](https://github.com/DSpace/dspace-angular/issues/2216#issuecomment-2360247323))
* Identify and extract the required modules from ng-dynamic-forms (core and ui-ng-bootstrap) and **add them to our dspace-angular codebase** (This would require retaining the code license on all files copied from ng-dynamic-forms
* Update component references in the existing codebase to point to the newly integrated modules.
* Remove the ng-dynamic-forms dependency from the project.

This essentially would provide several immediate benefits:
* Would remove our direct dependency on `ng-dynamic-forms` library (which is unmaintained)
* Would allow us to upgrade to `ng-bootstrap` v13, to help fix this accessibility ticket
* Would allow us to update the dependencies in the copied over `ng-dynamic-forms` code (to resolve other dependency issues)
* Would give us more time to come up with a full solution to #2216...as that will likely require a larger effort to rewrite our submission forms (and other forms) to no longer depend on the code from `ng-dynamic-forms`.

Even though it's never ideal to copy others code into DSpace, I approve of this approach as it seems to be the only way to solve our immediate accessibility issues _without a full scale redesign of our submission forms_.
